src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts

```diff
@@ -335,6 +349,52 @@ function attachInteropInvoker(): void {
   });
 }
 
+async function loadTimezone(timeZoneResource: LoadingResource) : Promise<void> {
+  const runDependencyId = `blazor:timezonedata`;
+  Module.addRunDependency(runDependencyId);
+  const request = await timeZoneResource.response;
+  let arrayBuffer = await request.arrayBuffer();
+
+  // The contents of the TZ file look like so
+  //
+  // [4 - byte length of manifest]
+  // [json manifest]
+  // [data bytes]
+  //
+  // The json manifest is an array that looks like so:
+  //
+  // [...["America/Fort_Nelson",2249],["America/Glace_Bay",2206]..]
+  //
+  // where the first token in each array is the relative path of the file on disk, and the second is the
+  // length of the file. The starting offset of a file can be calculated using the lengths of all files
+  // that appear prior to it.
+
+  const manifestSize = readInt32LE(new Uint8Array(arrayBuffer.slice(0, 4)), 0);
```

Comment:
Oh, hold on, `ArrayBuffer.slice` actually creates a separate copy of the buffer. So all these calls to `.slice` are very wasteful.

What I think you want instead here is:

```suggestion
  const manifestSize = readInt32LE(new Uint8Array(arrayBuffer, 0, 4));
```

The `Uint8Array` constructor produces a new view over the underlying buffer, but isn't copying anything.

Reply:
Gotcha. I'd thought this worked like `Span` or `ArraySegment`, strange that a slice produces copies